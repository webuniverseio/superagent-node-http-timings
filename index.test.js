const request = require('superagent');
const logNetworkTime = require('./index');
const https = require('https')
jest.setTimeout(10000);

const httpAgent = new https.Agent({keepAlive: true});
describe('GIVEN plugin was used WHEN response arrives THEN callback should get timings information', () => {
  test('timings', () => {
    const assertions = [
      'verified that timings were populated'
    ];
    expect.assertions(assertions.length);
    return expect(makeRequest()).resolves.toMatchSnapshot();
  });

  test('works with reusable keepAlive agent, we simply call the same request again with the same agent', () => {
    const assertions = [
      'verified that timings were populated'
    ];
    expect.assertions(assertions.length);
    return expect(makeRequest()).resolves.toMatchSnapshot();
  })

  function makeRequest() {
    return new Promise((res, rej) => {
      request
        .get(`https://google.com`)
        .use(logNetworkTime((err, result) => {
          if (err) {
            return rej(err)
          }
          res(valuesToTypes(result))
        }))
        .agent(httpAgent)
        .then(x => x)
        .catch(rej)
    })
  }
});

function valuesToTypes(obj) {
  return Object.keys(obj).reduce((acc, key) => {
    const value = obj[key];
    const type = typeof value;
    acc[key] = type === 'object' ? valuesToTypes(value) : type;
    return acc;
  }, {})
}
