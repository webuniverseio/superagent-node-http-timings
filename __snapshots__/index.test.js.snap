// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN plugin was used WHEN response arrives THEN callback should get timings information timings 1`] = `
Object {
  "status": "number",
  "timings": Object {
    "contentTransfer": "number",
    "dnsLookup": "number",
    "firstByte": "number",
    "socketAssigned": "number",
    "tcpConnection": "number",
    "tlsHandshake": "number",
    "total": "number",
  },
  "url": "string",
}
`;

exports[`GIVEN plugin was used WHEN response arrives THEN callback should get timings information works with reusable keepAlive agent, we simply call the same request again with the same agent 1`] = `
Object {
  "status": "number",
  "timings": Object {
    "contentTransfer": "number",
    "dnsLookup": "undefined",
    "firstByte": "number",
    "socketAssigned": "number",
    "tcpConnection": "undefined",
    "tlsHandshake": "undefined",
    "total": "number",
  },
  "url": "string",
}
`;
